ogzaf/demographics.py

```diff
@@ -104,83 +360,59 @@ def get_fert(totpers, min_yr, max_yr, graph=False):
             of life
 
     """
-    # Read raw data
-    pop_file = utils.read_file(
-        CUR_PATH, os.path.join("data", "demographic", "india_pop_data.csv")
+    # Get UN fertility rates for India for ages 15-49
+    ages_15_49 = np.arange(15, 50)
+    fert_rates_15_49 = (
+        get_un_fert_data(download=False)["fert_rate"].to_numpy().flatten()
     )
-    pop_data = pd.read_csv(pop_file, encoding="utf-8")
-    pop_data_samp = pop_data[
-        (pop_data["Age"] >= min_yr - 1) & (pop_data["Age"] <= max_yr - 1)
-    ]
-    age_year_all = pop_data_samp["Age"] + 1
-    curr_pop = np.array(pop_data_samp["2011"], dtype="f")
-    curr_pop_pct = curr_pop / curr_pop.sum()
-    # divide by 2000 because fertility rates per woman and we want per
-    # household
-    fert_data = (
-        np.array(
-            [
-                0.0,
-                1.0,
-                3.0,
-                10.7,
-                135.4,
-                166.0,
-                91.7,
-                32.7,
-                11.3,
-                4.1,
-                1.0,
-                0.0,
-            ]
-        )
-        / 2000
-    )
-    age_midp = np.array([9, 12, 15, 17, 22, 27, 32, 37, 42, 47, 52, 57])
-    # Generate interpolation functions for fertility rates
-    fert_func = si.interp1d(age_midp, fert_data, kind="cubic")
-    # Calculate average fertility rate in each age bin using trapezoid
-    # method with a large number of points in each bin.
-    binsize = (max_yr - min_yr + 1) / totpers
-    num_sub_bins = float(10000)
-    len_subbins = (np.float64(100 * num_sub_bins)) / totpers
-    age_sub = (
-        np.linspace(
-            np.float64(binsize) / num_sub_bins,
-            np.float64(max_yr),
-            int(num_sub_bins * max_yr),
+
+    # Extrapolate fertility rates for ages 1-14 and 50-100 using exponential
```

Comment:
Why are we extrapolating fertility rates for these age groups? The fertility for them is zero, empirically.